firebase-firestore/src/main/java/com/google/firebase/firestore/core/FirestoreClient.java

```diff
@@ -127,6 +114,21 @@ public FirestoreClient(
             throw new RuntimeException(e);
           }
         });
+
+    credentialsProvider.setChangeListener(
+        (User user) -> {
+          if (initialized.compareAndSet(false, true)) {
+            hardAssert(!firstUser.getTask().isComplete(), "Already fulfilled first user task");
+            firstUser.setResult(user);
+          } else {
+            // this fires twoce and sync engine isnt' initialized
```

Comment:
twoce => twice, isnt' => isn't

Btw does it even make sense to leave the comment here? After the fix, sync engine should be initialized even if this fires twice.